docs/tutorial/writing_plugins.md

```diff
@@ -0,0 +1,193 @@
+}
+```
+
+The class can have a `meta` property which provides the basic configuration for
+the plugin. It has the following fields: 
+- `name`: the name of the plugin
+- `version`: the version of the **library** the plugin is compatible with
+- `depends`: an array of strings describing other plugins this one depends on
+
+```typescript
+export class ExamplePlugin implements Plugins.Plugin {
+    meta = {
+        name: 'examplePlugin',
+        version: '0.9.1',
+        depends: []
+    }
+}
+```
+
+The main functionality of the plugin is provided by the `hooks` property. Use a
+getter function to define the object. There are three plugin hooks which can be
+defined in the returned object:
+
+- `prepareAnimation`: Called when the animate() parameters gets set in the
+  library to prepare the animation.
+- `registerAnimation`: Called when the animate() method called, and the lib
+  shedules the call to the animation queue.
+- `runAnimation`: Called when all animate() parameter set and animation can be
+  started. 
+
+Let’s add a simple function to the `prepareAnimation` phase. We’d like to
+automatically set the chart title to the name of the data series which is
+assigned to the color channel (if any). If no color is selected, then set a
+default title.
+
+### Adding hooks
+
+```typescript
+export class ExamplePlugin implements Plugins.Plugin {
+    get hooks(): Plugins.PluginHooks {
+        return {
+            prepareAnimation: (ctx: Plugins.PrepareAnimationContext, next: () => void): void => {
+                if ('config' in ctx.target) {
+                    ctx.target.config.title = ctx.target.config?.color ?? 'Example plugin'
+                }
+                next()
+            }
+        }
+    }
+}
+```
+
+The hooks can also have a `priority` number that decides the order the
+registered plugins run. It’s a float value, ranging from 0 (first) to 1 (last).
+
+```typescript
+export class ExamplePlugin implements Plugins.Plugin {
+    get hooks(): Plugins.PluginHooks {
+        return {
+            prepareAnimation: Object.assign(
+                (ctx: Plugins.PrepareAnimationContext, next: () => void): void => {
+					if ('config' in ctx.target) {
+						ctx.target.config.title = ctx.target.config?.color ?? 'Example plugin'
+					}
+					next()
+				},
+                { priority: 1 }
+            )
+        }
+    }
+}
+```
+
+### Adding API functions
+
+The class can also define an `api` property. It can contain one or more
+functions which can be called from the original context using
+`chart.feature.<plugin name>.<function name>()`.
+
+Let’s add an API function to our plugin which can tell if the default title has
+been changed.
+
+```typescript
+import { type Plugins } from 'vizzu'
+
+interface ExamplePluginApi extends Plugins.PluginApi {
+    hasDefaultTitle(): boolean
+}
+
+export class ExamplePlugin implements Plugins.Plugin {
+	meta: Plugins.PluginMeta = {
+		name: 'examplePlugin',
+		version: '0.9.1',
```

Comment:
there is still a fix lib version in this place

Reply:
we use this script to replace the latest URLs with versioned URLs:
https://github.com/vizzuhq/vizzu-lib/blob/plugin-howto/tools/modules/vizzu.py
we run set_version on the tutorial markdown files, so this script can be used for example, to replace the "version: 'latest'" string with the correct version.

But as I see this version replacement is bugged since we reorganize the how to setup parts of the tutorial, I will fix it soon